### PR TITLE
handle errors on 200 responses

### PR DESCRIPTION
**Pull request by dmfenton**
ping @ngoldman

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -404,84 +404,85 @@ FeatureService.prototype._requestFeatures = function (task, cb) {
...
-    }
-    // immediately kill
-    if (!task.retry) {
-      task.retry = 1
+/* Decodes a response for features
+ * @param {object} res - the response received from the GIS Server
+ * @param {array} data - an array of chunks received from the server
+ * @param {function} callback - calls back with either an error or the decoded feature json
+ */
+FeatureService.prototype._decode = function (res, data, callback) {
```
Comment (ungoldman) [inline index.js:442]: `_decode` here does not match `decode` above
https://github.com/koopjs/featureservice/pull/9/files#diff-168726dbe96b3ce427e7fedce31bb0bcR411

Comment (dmfenton) [inline index.js:442]: great catch, now how the heck are my tests passing?

Comment (ungoldman) [inline index.js:442]: they're testing the `_decode` method directly afaict

**Comment by ungoldman** [comment]
seems aight

**Pull request merged**